extensions/json-jackson/src/main/java/io/deephaven/json/jackson/Parsing.java

```diff
@@ -0,0 +1,303 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json.jackson;
+
+import ch.randelshofer.fastdoubleparser.JavaBigDecimalParser;
+import ch.randelshofer.fastdoubleparser.JavaBigIntegerParser;
+import ch.randelshofer.fastdoubleparser.JavaDoubleParser;
+import ch.randelshofer.fastdoubleparser.JavaFloatParser;
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.StreamReadFeature;
+import io.deephaven.util.BooleanUtils;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.CharBuffer;
+
+final class Parsing {
+
+    static void assertNoCurrentToken(JsonParser parser) {
+        if (parser.hasCurrentToken()) {
+            throw new IllegalStateException(
```

Comment:
These exceptions suggest we may want to catch them and add context at a higher level.

Reply:
The introduction of a `ValueAwareException` allows higher levels to wrap these lower level errors with much more context.